datetime/datetime_test.go

```diff
@@ -0,0 +1,543 @@
+		t.Fatalf("Failed to SelectTyped(): %s", err.Error())
+	}
+
+	// Teardown: delete a value.
+	_, err = conn.Delete(space, index, []interface{}{tm})
+	if err != nil {
+		t.Fatalf("Datetime delete failed: %s", err.Error())
+	}
+}
+
+func TestCustomEncodeDecodeTuple3(t *testing.T) {
+	if isDatetimeSupported == false {
+		t.Skip("Skipping test for Tarantool without datetime support in msgpack")
+	}
+
+	conn := connectWithValidation(t)
+	defer conn.Close()
+
+	// Setup: insert a value.
+	tm := time.Unix(0, 0)
+	_, err := conn.Insert(space, []interface{}{tm})
+	if err != nil {
+		t.Fatalf("Datetime insert failed: %s", err.Error())
+	}
+
+	var tuples2 []Tuple2
+	err = conn.SelectTyped("testDatetime_2", index, 0, 1, IterEq, []interface{}{1, tm}, &tuples2)
+	if err != nil {
+		log.Fatalf("Failed to SelectTyped: %s", err.Error())
+		return
+	}
+
+	// Teardown: delete a value.
+	_, err = conn.Delete(space, index, []interface{}{tm})
+	if err != nil {
+		t.Fatalf("Datetime delete failed: %s", err.Error())
+	}
+}
+
+func TestCustomDecodeFunction(t *testing.T) {
+	if isDatetimeSupported == false {
+		t.Skip("Skipping test for Tarantool without datetime support in msgpack")
+	}
+
+	conn := connectWithValidation(t)
+	defer conn.Close()
+
+	// Call function 'call_me_maybe' returning a table of custom tuples.
+	var tuples []Tuple
+	err := conn.CallTyped("call_me_maybe", []interface{}{1}, &tuples)
+	if err != nil {
+		t.Fatalf("Failed to CallTyped: %s", err.Error())
+	}
+
+	if cid := tuples[0].Cid; cid != 5 {
+		t.Fatalf("Wrong Cid (%d), should be 5", cid)
+	}
+	if orig := tuples[0].Orig; orig != "Poyekhali!" {
+		t.Fatalf("Wrong Orig (%s), should be 'Hello, there!'", orig)
+	}
+}
+
+func TestCustomEncodeDecodeTuple5(t *testing.T) {
+	if isDatetimeSupported == false {
+		t.Skip("Skipping test for Tarantool without datetime support in msgpack")
+	}
+
+	conn := connectWithValidation(t)
+	defer conn.Close()
+
+	dt := time.Unix(500, 1000)
+	_, err := conn.Insert(space, []interface{}{dt})
+	if err != nil {
+		t.Fatalf("Datetime insert failed: %s", err.Error())
+	}
+
+	resp, errSel := conn.Select(space, index, 0, 1, IterEq, []interface{}{dt})
+	if errSel != nil {
+		t.Errorf("Failed to Select: %s", errSel.Error())
+	}
+	if tpl, ok := resp.Data[0].([]interface{}); !ok {
+		t.Errorf("Unexpected body of Select")
+	} else {
+		if val, ok := tpl[0].(time.Time); !ok || !val.Equal(dt) {
+			t.Fatalf("Unexpected body of Select")
+		}
+	}
+
+	// Teardown: delete a value.
+	_, err = conn.Delete(space, index, []interface{}{dt})
+	if err != nil {
+		t.Fatalf("Datetime delete failed: %s", err.Error())
+	}
+}
+
+// runTestMain is a body of TestMain function
+// (see https://pkg.go.dev/testing#hdr-Main).
+// Using defer + os.Exit is not works so TestMain body
+// is a separate function, see
+// https://stackoverflow.com/questions/27629380/how-to-exit-a-go-program-honoring-deferred-calls
+func runTestMain(m *testing.M) int {
+	isLess, err := test_helpers.IsTarantoolVersionLess(2, 10, 0)
```

Comment:
Should we say something about these limitations in README.md or CONTRIBUTING.md?

Should we add 2.10 tarantool to `.github/workflows/testing.yml`? I think 2.x-latest is not enough. Уou can replace 2.9 with 2.10 there.

Reply:
> Should we say something about the limitation in README.md or CONTRIBUTING.md?

What limitation you are talking about?

> Should we add 2.10 tarantool to .github/workflows/testing.yml?
> I think 2.x-latest is not enough.

Could you explain why do you think so?

Reply:
> > Should we say something about the limitation in README.md or CONTRIBUTING.md?
> 
> What limitation you are talking about?

At least Tarantool 2.10 is required for the datetime support.

> > Should we add 2.10 tarantool to .github/workflows/testing.yml?
> > I think 2.x-latest is not enough.
> 
> Could you explain why do you think so?

Because `2.x-latest` will became `2.12` or `2.14` in the future and we'll miss 2.10.

Reply:
>>> Should we say something about the limitation in README.md or CONTRIBUTING.md?

>> What limitation you are talking about?

> At least Tarantool 2.10 is required for the datetime support.

There a Godoc for such purposes.

> Should we add 2.10 tarantool to .github/workflows/testing.yml?

Ok, added.